src/python/dxpy/utils/executable_unbuilder.py

```diff
@@ -115,85 +116,119 @@ def make_cluster_bootstrap_script_file(region, entry_point, code, suffix):
+    
+    # Get regions where the user's billTo are permitted
+    try:
+        bill_to = dxpy.api.user_describe(dxpy.whoami())['billTo']
+        permitted_regions = set(dxpy.DXHTTPRequest('/' + bill_to + '/describe', {}).get("permittedRegions"))
+    except DXError:
+        print("Failed to get permitted regions where {} can perform billable activities.\n".format(bill_to), file=sys.stderr)
+        sys.exit(1)    
+    
+    enabled_regions = set(info["runSpec"]["bundledDependsByRegion"].keys())
+    if not enabled_regions.issubset(permitted_regions):
+        print("Region(s) {} are not among the permitted regions of {}. Resources from these regions will not be available.".format(
+            ", ".join(enabled_regions.difference(permitted_regions)), bill_to))
+    # Update enabled regions
+    enabled_regions.intersection_update(permitted_regions)
+    # Check if at least one region is enabled when not omitting resources
     if not omit_resources:
-        for dep in info["runSpec"]["bundledDepends"]:
-            if dep in deps_to_remove:
-                continue
-            handler = get_handler(dep["id"])
-            if isinstance(handler, dxpy.DXFile):
+        if not enabled_regions:
+            raise DXError(
+                "Cannot download resources of the requested executable {} since it is not available in any of the billable regions. "
+                "You can use the --omit-resources flag to skip downloading the resources. ".format(info["name"]))
+        
+        print("Dependencies could be retrieved from region: {}. ".format(", ".join(enabled_regions)))
+        # Pick a source region. The current selected region is preferred
+        try:
+            current_region = dxpy.api.project_describe(dxpy.WORKSPACE_ID, input_params={"fields": {"region": True}})["region"]
+        except:
+            current_region = None
+
+        if current_region in enabled_regions:
+            source_region  = current_region
+            print("Trying to download resources from the current region {}...".format(source_region))
+        else:
+            source_region = enabled_regions.pop()
+            print("Trying to download resources from one of the enabled region {}...".format(source_region))
+
+        # When an applet is built bundledDepends are added in the following order:
+        # 1. bundledDepends explicitly specified in the dxapp.json
+        # 2. resources (contents of resources directory added as bundledDepends)
+        # 3. assetDepends (translated into bundledDepends)
+        #
+        # Therefore while translating bundledDepends to assetDepends, we are traversing the
+        # list in reverse order and exiting when we can't find the "AssetBundle" property
+        # with the tarball file.
+        #
+        # NOTE: If last item (and contiguous earlier items) of bundledDepends (#1 above) refers to an
+        # AssetBundle tarball, those items will be converted to assetDepends.
```

Comment:
I think think comments needs to be updated since we are not trying to extract and reconstruct assets in dxapp.json any more.

Would you add to this comment what you say here: https://github.com/dnanexus/dx-toolkit/pull/717/files#r731166369 That assetDepends are unpacked to bundledDepends in dxapp.json?

One more thing: Did you try to build a sequence: dx get, dx build, dx get? It may be tricky and out of scope for this task to get it completely right so just let me know if this causes issues under this new design/code.